nixos/modules/services/networking/croc.nix

```diff
@@ -0,0 +1,78 @@
+{ config, lib, pkgs, ... }:
+let
+  inherit (lib) types;
+  cfg = config.services.croc;
+in
+{
+  options.services.croc = {
+    enable = lib.mkEnableOption "croc relay";
+    ports = lib.mkOption {
+      type = types.listOf types.port;
+      default = [9009 9010 9011 9012 9013];
+      description = "Ports of the relay.";
+    };
+    pass = lib.mkOption {
+      type = types.str;
+      default = "pass123";
+      description = "Password for the relay (warning: it will be visible in the Nix store and the list of processes).";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.services.croc = {
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        ExecStart = "${pkgs.croc}/bin/croc --pass '${cfg.pass}' relay --ports ${lib.concatMapStringsSep "," toString cfg.ports}";
+        # systemd-analyze security croc
+        # → Overall exposure level for croc.service: 1.1 OK
+        AmbientCapabilities = "";
+        CapabilityBoundingSet = "";
+        DynamicUser = true;
+        LockPersonality = true;
+        MemoryDenyWriteExecute = true;
+        NoNewPrivileges = true;
+        PrivateDevices = true;
+        PrivateMounts = true;
+        PrivateNetwork = false;
+        PrivateTmp = true;
```

Comment:
Not that it hurts but I'm pretty sure at least one or two of these options are redundant given `DynamicUser` (and maybe others). Since I'm not providing you specifics feel free to either do a quick search and see or disregard this comment :smile:

Reply:
Indeed, this is not the smallest config possible to get an equivalent security, but this list of options matches the report of `systemd-analyze security $service` and I'm using it systematically because:
1. well.. securing services is already too much time consuming for me (eg. `SystemCallFilter=`) so I pass on polishing the options :P
2. and also to avoid any surprise (eg. if one day `DynamicUser=` is removed but the security is not restricted accordingly).

Reply:
That's a pretty solid argument. Thanks for sharing that, I'll try the same next time I attempt to harden a service.